### DISCUSSION TITLE
Select component custom key addition for loop?

### DISCUSSION THREAD

**Discussion by nayeshdaggula** [mantinedev/mantine#2730] [Q&A]
hi

is there any possible to add custom key for loop of the array in select component? i have country code list but most of the values similar in country code with string getting error on console duplicated keys

Thank you

**Comment by rtivital** [comment]
No, that is not supported, `value` field for select items is required to unique.

**Comment by nayeshdaggula** [comment]
yes i know that but if you guy can give option to add key in the custom item component. maybe it will solve the issue

**Reply by rtivital** [reply]
While `value` is required to be unique, other fields are not. With custom item component you can structure your data to include any amount of duplicated fields.

**Reply by nayeshdaggula** [reply]
no problem. its countries phone codes data so not possible to have unique value

Thank you for information

**Reply by zhulien-ivanov** [reply]
> no problem. its countries phone codes data so not possible to have unique value
> 
> Thank you for information

Phone codes should be unique in the first place, that's the whole point. Aren't they bound to country? In any case, if they're not for some reason, simply concat the country code before the phone.

**Reply by rtivital** [reply]
I believe there are several countries that have the same phone code, for example, Russia and Kazakhstan share `+7` code
